### PR TITLE
Missing parameter description in `morphology.reconstruction() docstring #3581

### PR DESCRIPTION
**Pull request by FrAnCOisCokELaER**
## Description
[Missing parameter description in `morphology.reconstruction() docstring #3581
_I have completed the description for parameters 'method' and 'offset'.
_I have added 2 conditions that seems to be fulfilled for the 'offset' parameter when not default.
_On the previous revision when 'offset' was not default, nothing prevented 'selem' to be non-odd and the padding could also possibly not be symmetric but seemed to be not handled. I have added a generic way to handle the padding.
_ Two tests were added: one for checking a non-valid offset and one for checking the vailidity of a result with a non-symmetric padding.
_Travis build is ok on the branch  ]

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](https://scikit-image.org/docs/dev/contribute.html)]

## References
[If this is a bug-fix or enhancement, it closes issue # ]
[If this is a new feature, it implements the following paper: ]

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by emmanuelle** [comment]
Thank you @FrAnCOisCokELaER ! For the case you fixed with the padding, would it be possible to add a test which breaks on current master but passes with your changes?

File: skimage/morphology/greyreconstruct.py [hunk 1]
```diff
@@ -139,15 +144,20 @@ def reconstruction(seed, mask, method='dilation', selem=None, offset=None):
...
+            raise ValueError("Offset must be included inside selem")
+
     # Cross out the center of the selem
     selem[tuple(slice(d, d + 1) for d in offset)] = False
 
     # Make padding for edges of reconstructed image so we can ignore boundaries
-    padding = (np.array(selem.shape) / 2).astype(int)
     dims = np.zeros(seed.ndim + 1, dtype=int)
-    dims[1:] = np.array(seed.shape) + 2 * padding
+    dims[1:] = np.array(seed.shape) + (np.array(selem.shape) - 1).astype(int)
```
Comment (sciunto) [inline skimage/morphology/greyreconstruct.py]: Is it still necessary to cast the type? (here and below)

**Comment by sciunto** [comment]
Great job.

I'm +1 with Emmanuelle.

In addition, I think I discovered an inconsistency in the docstring. The rest is minor...

File: skimage/morphology/greyreconstruct.py (hunk 1 above)
Comment (FrAnCOisCokELaER) [reply skimage/morphology/greyreconstruct.py]: No of course you're right , the fix is in https://github.com/scikit-image/scikit-image/pull/3591/commits/2d0f9fa2bdf1844a63fa92e709a4407d966790a5

File: skimage/morphology/greyreconstruct.py [hunk 2]
```diff
@@ -42,15 +42,20 @@ def reconstruction(seed, mask, method='dilation', selem=None, offset=None):
...
         to the corresponding mask value; for erosion, the reverse is true.
-    selem : ndarray
+        Default is 'dilation'.
+    selem : ndarray, optional
         The neighborhood expressed as an n-D array of 1's and 0's.
         Default is the ball of radius 1 according to the maximum norm
         (i.e. a 3x3 square for 2D images, a 3x3x3 cube for 3D images, etc.)
+    offset : ndarray, optional
+        The coordinates of the center of the structuring element.
+        Default is located on the real center of the selem, in that case
```
[the remark this answers is not in the document]
Comment (FrAnCOisCokELaER) [reply skimage/morphology/greyreconstruct.py]: yes seems clearer , the fix in https://github.com/scikit-image/scikit-image/pull/3591/commits/d97191261b6204130e7d1b4105d8ab8da9827d46

File: skimage/morphology/greyreconstruct.py [hunk 3]
```diff
@@ -42,15 +42,20 @@ def reconstruction(seed, mask, method='dilation', selem=None, offset=None):
...
         erosion), the seed image is dilated (or eroded) until limited by the
         mask image. For dilation, each seed value must be less than or equal
         to the corresponding mask value; for erosion, the reverse is true.
-    selem : ndarray
+        Default is 'dilation'.
+    selem : ndarray, optional
         The neighborhood expressed as an n-D array of 1's and 0's.
-        Default is the ball of radius 1 according to the maximum norm
-        (i.e. a 3x3 square for 2D images, a 3x3x3 cube for 3D images, etc.)
+        Default is the n-D square of radius equal to 1 (i.e. a 3x3 square
```
Comment (emmanuelle) [inline skimage/morphology/greyreconstruct.py]: I'm not sure it's correct to speak about a square of radius 1, how about the square of length 3?

File: skimage/morphology/greyreconstruct.py [hunk 4]
```diff
@@ -42,15 +42,20 @@ def reconstruction(seed, mask, method='dilation', selem=None, offset=None):
...
     mask : ndarray
         The maximum (dilation) / minimum (erosion) allowed value at each pixel.
-    method : {'dilation'|'erosion'}
+    method : {'dilation'|'erosion'}, optional
         Perform reconstruction by dilation or erosion. In dilation (or
         erosion), the seed image is dilated (or eroded) until limited by the
         mask image. For dilation, each seed value must be less than or equal
         to the corresponding mask value; for erosion, the reverse is true.
-    selem : ndarray
+        Default is 'dilation'.
```
Comment (jni) [inline skimage/morphology/greyreconstruct.py]: Typically we don't include the defaults here when they are available from the function signature...

**Pull request merged**